File: crates/forge-ast/tests/fixtures/typescript/auth_service.ts
import { readFileSync } from "fs";

export const TOKEN_TTL_SECONDS = 3600;

export interface TokenVerifier {
  verify(token: string): boolean;
}

export class AuthService {
  constructor(private store: string) {}

  authenticate(token: string): boolean {
    return validateToken(token);
  }
}

export function validateToken(token: string): boolean {
  return token.length > 0;
}

export type AuthResult = boolean;

readFileSync("auth.txt");

File: crates/forge-ast/tests/fixtures/rust/auth_service.rs
use std::sync::Arc;

pub mod nested;

pub const TOKEN_TTL_SECONDS: u64 = 3600;

pub static AUTH_HEADER: &str = "authorization";

pub struct AuthService {
    store: Arc<String>,
}

pub enum AuthError {
    Missing,
    Invalid,
}

pub trait TokenVerifier {
    fn verify(&self, token: &str) -> bool;
}

impl AuthService {
    pub fn new(store: Arc<String>) -> Self {
        Self { store }
    }

    pub fn authenticate(&self, token: &str) -> Result<(), AuthError> {
        validate_token(token);
        Ok(())
    }
}

pub fn validate_token(token: &str) -> bool {
    !token.is_empty()
}
